boa_engine/src/builtins/temporal/duration/mod.rs

```diff
@@ -698,7 +698,107 @@ impl Duration {
             validate_temporal_rounding_increment(rounding_increment, f64::from(max), false)?;
         }
 
-        // TODO: Complete the rest of the new `Temporal.Duration.prototype.round` impl.
+        // 25. Let hoursToDaysConversionMayOccur be false.
+        // 26. If duration.[[Days]] ≠ 0 and zonedRelativeTo is not undefined, set hoursToDaysConversionMayOccur to true.
+        // 27. Else if abs(duration.[[Hours]]) ≥ 24, set hoursToDaysConversionMayOccur to true.
+        let conversion_may_occur = if duration.inner.days() != 0.0 && zoned_relative_to.is_some() {
+            true
+        } else {
+            24f64 <= duration.inner.hours().abs()
+        };
+
+        // 28. If smallestUnit is "nanosecond" and roundingIncrement = 1, let roundingGranularityIsNoop be true; else let roundingGranularityIsNoop be false.
+        let is_noop = smallest_unit == TemporalUnit::Nanosecond
+            && (rounding_increment - 1f64).abs() < f64::EPSILON;
```

Comment:
I'd suggest using an `u16` for the rounding increment insteaed to avoid having to use `f64` here. This is what is used on `get_digit_format_options`.

Reply:
The one thing that's a bit rough is that the spec currently has `RoundNumberToIncrement` and `RoundNumberToIncrementAsIfPositive`, so `rounding_increment` can be negative in this instance (granted that could change depending on `Intl.NumberFormat V3`).

Reply:
I don't think it can though? I'm pretty sure what could be negative is `x`, the number being formatted. `rounding_increment` is strictly positive thanks to the check inside `https://tc39.es/proposal-temporal/#sec-temporal-totemporalroundingincrement`

Reply:
Yep, you're 100% right. I'm going to have to keep track of this change, but I think it should be fine.